module/VuFind/src/VuFind/Db/Service/UserService.php

```diff
@@ -110,7 +130,13 @@ public function deleteUser(UserEntityInterface|int $userOrId): void
      */
     public function getUserById(int $id): ?UserEntityInterface
     {
-        return $this->entityManager->find($this->getEntityClass(User::class), $id);
+        $dql = 'SELECT u '
+                . 'FROM ' . $this->getEntityClass(UserEntityInterface::class) . ' u '
+                . 'WHERE u.id = :id';
+        $query = $this->entityManager->createQuery($dql);
+        $query->setParameter('id', $id);
+        $result = $query->getResult();
```

Comment:
Is `getResult` going to return an array of results? If so, we might want to `return $result[0] ?? null;` instead of returning the whole array.

Reply:
Since id here is the primary key, I am expecting the query to return a single object. Please correct me if my understanding is wrong!

Reply:
You are correct -- I misread `getResult` as `getResults`. :-)